### PR TITLE
feat: add chat input, hook frontend to backend

### PR DESCRIPTION
**Pull request by michelle-pine**
Adds the chat input! Please note that i can't run tests on the pause/resume game functionality. 
image

Messages can now be successfully sent and received from the frontend/backend.

### CODE REVIEW

File: frontend/src/components/world/WorldMap.tsx [hunk 1]
```diff
@@ -189,7 +189,7 @@ class CoveyGameScene extends Phaser.Scene {
         !this.lastLocation ||
         this.lastLocation.x !== body.x ||
         this.lastLocation.y !== body.y ||
-        this.lastLocation.rotation !== primaryDirection ||
+        (isMoving && this.lastLocation.rotation !== primaryDirection) ||
```
Comment (michelle-pine) [inline frontend/src/components/world/WorldMap.tsx:192]: image

File: frontend/src/components/Chat/ChatContainer.tsx [hunk 2]
```diff
@@ -19,10 +20,10 @@ const sampleMessage: Message = {
   },
   directMessageId: null,
 };
-
+ 
 interface ChatContainerProps {
   // so for now- Town/Proximity will pass in undefined
-  directMessageID: string | undefined;
+  directMessageID: string | null;
```
Comment (anikarabenhorst) [inline frontend/src/components/Chat/ChatContainer.tsx:26]: everytime i see a change from undefined to null, a part of me dies

**Review by MahemaS27 — APPROVED** [review]
frick iSmoving. But LGTM!

**Review by anikarabenhorst — APPROVED** [review]
woo looks good

**Pull request merged**